homeassistant/components/statistics/sensor.py

```diff
@@ -195,27 +221,82 @@ def _add_state_to_queue(self, new_state):
             )
             return
 
-        self._unit_of_measurement = new_state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
+        self._unit_of_measurement = self._derive_unit_of_measurement(new_state)
```

Comment:
@emontnemery you were the one to move the this into `_add_state_to_queue` in https://github.com/home-assistant/core/pull/58023 Is deriving the unit for every new state value really necessary?

Reply:
Nice. I can still learn some things from you :)

Reply:
I gave this a shot but as far as I got my conclusion is that I would need to convert the function into a staticmethod. Not a total showstopper but feels like it's not worth it for this arguably simple function. Will leave as is.

Reply:
Yes, you're absolutely right about that.
We would have to use some 3rd party library to have it work as expected.